ibr/ibr-sdk/src/renderer.js

```diff
@@ -10,49 +10,59 @@
  */
 function renderLayer(data) {
 
-    // Extract coordinates range from ibr data
-    var deserializedData = InternalBuildingRepresentation.read( new Pbf( data ) );
-    var coordsIndexList = deserializedData.visualization[0].coordinates;
-    var coordsRangeBuffer = coordsIndexList.buffer.slice( coordsIndexList.byteOffset, coordsIndexList.buffer.byteLength );
-    var coordsRange = new Uint32Array( coordsRangeBuffer );
-    for (var i = 0; i < coordsRange.length; i++) {
-        coordsRange[i] = swap32( coordsRange[i] );
+    // Decode Indices from data.visualization[].coordinate_indices
+    var deserializedData = InternalBuildingRepresentation.read(new Pbf(data));
+    var coordsIndexList, coordsRangeBuffer, coordsRange;
+    var coordsRangeList = [];
+    for (const visLayer of deserializedData.visualization) {
+        coordsIndexList = visLayer.coordinate_indices;
+        coordsRangeBuffer = coordsIndexList.buffer.slice(coordsIndexList.byteOffset, coordsIndexList.byteOffset+coordsIndexList.length);
+        coordsRange = new Uint32Array(coordsRangeBuffer);
+        for (var i = 0; i < coordsRange.length; i++) {
+            coordsRange[i] = swap32(coordsRange[i]);
+        }
+        coordsRangeList.push(coordsRange);
     }
 
-    // Decode coordinates from data.coordinatesLookup.encodedData
-    var decoder = new TextDecoder('utf8');
-    var decodedCoordsString = atob( decoder.decode( deserializedData.coordinates_lookup.encoded_data ) );
-
-    // Put coordinates in data structure
-    var coords = decodedCoordsString.split(",");
-    var decodedCoordsList = [];
-    for (const coord of coords) {
-        var coordinate = coord.split(" ");
-        if (coordinate.length === 2) {
-            decodedCoordsList.push( coordinate );
-        }
+    // Decode Coordinates from data.coordinates_lookup
+    var coordsLookup = deserializedData.coordinates_lookup;
+    var coordsLookupBuffer = coordsLookup.buffer.slice(coordsLookup.byteOffset, coordsLookup.byteOffset+coordsLookup.length);
+    var coordsLookupDV = new DataView(coordsLookupBuffer);
+    var coordsLookupList = [];
+    for (var i = 0; i < coordsLookup.length/4; i+=4) {
```

Comment:
nit: 4 refers to bytes per element? let's reference [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/BYTES_PER_ELEMENT) instead of having a magic int